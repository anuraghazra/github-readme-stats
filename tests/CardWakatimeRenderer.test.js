import "@testing-library/jest-dom";
import WakaTimeCard from "../src/cards/wakatime";
import {
  genWakatimeMockData,
  mockVercel,
  mockWakaTimeRequest,
} from "./utils/mock";

describe("Test Render Wakatime Card", () => {
  async function render({ query = {}, stats = {} } = {}) {
    const { req, res } = mockVercel(query);
    const mockRestore = mockWakaTimeRequest({
      ...genWakatimeMockData(),
      ...stats,
    });
    const svgString = await new WakaTimeCard(req.query).generateSvgString(
      res.setHeader,
    );
    mockRestore();
    return svgString;
  }
  it("should render correctly", async () => {
    const svgString = await render();
    expect(svgString).toMatchSnapshot();
  });

  it("should render correctly with compact layout", async () => {
    const card = await render({
      query: {
        layout: "compact",
      },
    });
    expect(card).toMatchSnapshot();
  });

  it("should render translations", async () => {
    document.body.innerHTML = await render({
      query: {
        locale: "cn",
      },
      stats: {
        data: {},
      },
    });
    expect(document.getElementsByClassName("header")[0].textContent).toBe(
      "Wakatime 周统计",
    );
    expect(
      document.querySelector('g[transform="translate(0, 0)"]>text.stat.bold')
        .textContent,
    ).toBe("本周没有编程活动");
  });

  it("should render without rounding", async () => {
    document.body.innerHTML = await render({
      query: {
        border_radius: "0",
      },
    });
    expect(document.querySelector("rect")).toHaveAttribute("rx", "0");
    document.body.innerHTML = await render();
    expect(document.querySelector("rect")).toHaveAttribute("rx", "4.5");
  });
});
