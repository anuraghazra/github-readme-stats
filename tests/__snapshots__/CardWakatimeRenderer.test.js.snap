// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Render Wakatime Card should render correctly 1`] = `
"<svg width=\\"495\\" height=\\"150\\" viewBox=\\"0 0 495 150\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><style>
        
      .font-bold { font-weight: 700 }
      .fadeIn {
        opacity: 0;
        animation: fadeInAnimation 0.3s ease-in-out forwards;
      }
  
      .icon {
        fill: #4c71f2;
      }
      .primary-stroke {
        stroke: #2f80ed
      }
      .primary-fill {
        fill: #2f80ed
      }
      .text-fill {
        fill: #333
      }
      .text-stroke {
        stroke: #333
      }

    
    
        
      </style><rect data-testid=\\"card-bg\\" x=\\"0.5\\" y=\\"0.5\\" rx=\\"4.5\\" height=\\"99%\\" stroke=\\"#e4e2e2\\" width=\\"494\\" fill=\\"#fffefe\\" stroke-opacity=\\"1\\"></rect><g data-testid=\\"card-title\\" transform=\\"translate(25, 35)\\"><g transform=\\"translate(0, 0)\\"><text x=\\"0\\" y=\\"0\\" data-testid=\\"header\\" style=\\"animation:fadeInAnimation 0.8s ease-in-out forwards;font:600 18px 'Segoe UI', Ubuntu, Sans-Serif\\" class=\\"primary-fill\\">Wakatime Stats</text></g></g><g data-testid=\\"main-card-body\\" transform=\\"translate(0, 55)\\"><svg x=\\"25\\" y=\\"0\\" width=\\"100%\\"><style></style><g transform=\\"translate(0, 0)\\"><g class=\\"fadeIn\\" style=\\"animation-delay: 450ms\\"><text class=\\"font-bold text-fill\\" y=\\"12.5\\" style=\\"font:600 14px 'Segoe UI', Ubuntu, 'Helvetica Neue', Sans-Serif\\">Other:</text><text style=\\"font:600 14px 'Segoe UI', Ubuntu, 'Helvetica Neue', Sans-Serif\\" class=\\"font-bold text-fill\\" x=\\"350\\" y=\\"12.5\\" data-testid=\\"Other\\">19 mins</text><g data-testid=\\"undefined\\" transform=\\"translate(110, 4)\\" style=\\"font:400 11px 'Segoe UI', Ubuntu, Sans-Serif\\" class=\\"text-fill\\"><svg width=\\"220\\" x=\\"0\\" y=\\"0\\"><style></style><rect rx=\\"5\\" ry=\\"5\\" x=\\"0\\" y=\\"0\\" width=\\"220\\" height=\\"8\\" fill=\\"#333\\"></rect><rect height=\\"8\\" fill=\\"#2f80ed\\" rx=\\"5\\" ry=\\"5\\" x=\\"0\\" y=\\"0\\" data-testid=\\"progress\\" width=\\"93.46%\\"></rect></svg></g></g></g><g transform=\\"translate(0, 25)\\"><g class=\\"fadeIn\\" style=\\"animation-delay: 600ms\\"><text class=\\"font-bold text-fill\\" y=\\"12.5\\" style=\\"font:600 14px 'Segoe UI', Ubuntu, 'Helvetica Neue', Sans-Serif\\">TypeScript:</text><text style=\\"font:600 14px 'Segoe UI', Ubuntu, 'Helvetica Neue', Sans-Serif\\" class=\\"font-bold text-fill\\" x=\\"350\\" y=\\"12.5\\" data-testid=\\"TypeScript\\">1 min</text><g data-testid=\\"undefined\\" transform=\\"translate(110, 4)\\" style=\\"font:400 11px 'Segoe UI', Ubuntu, Sans-Serif\\" class=\\"text-fill\\"><svg width=\\"220\\" x=\\"0\\" y=\\"0\\"><style></style><rect rx=\\"5\\" ry=\\"5\\" x=\\"0\\" y=\\"0\\" width=\\"220\\" height=\\"8\\" fill=\\"#333\\"></rect><rect height=\\"8\\" fill=\\"#2f80ed\\" rx=\\"5\\" ry=\\"5\\" x=\\"0\\" y=\\"0\\" data-testid=\\"progress\\" width=\\"6.54%\\"></rect></svg></g></g></g></svg></g></svg>"
`;

exports[`Test Render Wakatime Card should render correctly with compact layout 1`] = `
"<svg width=\\"495\\" height=\\"115\\" viewBox=\\"0 0 495 115\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><style>
        
      .font-bold { font-weight: 700 }
      .fadeIn {
        opacity: 0;
        animation: fadeInAnimation 0.3s ease-in-out forwards;
      }
  
      .icon {
        fill: #4c71f2;
      }
      .primary-stroke {
        stroke: #2f80ed
      }
      .primary-fill {
        fill: #2f80ed
      }
      .text-fill {
        fill: #333
      }
      .text-stroke {
        stroke: #333
      }

    
    
        
      </style><rect data-testid=\\"card-bg\\" x=\\"0.5\\" y=\\"0.5\\" rx=\\"4.5\\" height=\\"99%\\" stroke=\\"#e4e2e2\\" width=\\"494\\" fill=\\"#fffefe\\" stroke-opacity=\\"1\\"></rect><g data-testid=\\"card-title\\" transform=\\"translate(25, 35)\\"><g transform=\\"translate(0, 0)\\"><text x=\\"0\\" y=\\"0\\" data-testid=\\"header\\" style=\\"animation:fadeInAnimation 0.8s ease-in-out forwards;font:600 18px 'Segoe UI', Ubuntu, Sans-Serif\\" class=\\"primary-fill\\">Wakatime Stats</text></g></g><g data-testid=\\"main-card-body\\" transform=\\"translate(0, 55)\\"><svg x=\\"25\\" y=\\"0\\" width=\\"100%\\"><style></style><g><mask id=\\"rect-mask\\"><rect x=\\"0\\" y=\\"0\\" width=\\"440\\" height=\\"8\\" fill=\\"white\\" rx=\\"5\\"></rect></mask><rect mask=\\"url(#rect-mask)\\" data-testid=\\"lang-progress\\" x=\\"0\\" y=\\"0\\" width=\\"411.22\\" height=\\"8\\" fill=\\"#858585\\"></rect><rect mask=\\"url(#rect-mask)\\" data-testid=\\"lang-progress\\" x=\\"411.22\\" y=\\"0\\" width=\\"28.78\\" height=\\"8\\" fill=\\"#2b7489\\"></rect><g data-testid=\\"undefined\\" transform=\\"translate(0, 25)\\"><circle data-testid=\\"lang-color\\" cx=\\"5\\" cy=\\"6\\" r=\\"5\\" fill=\\"#858585\\"></circle><text data-testid=\\"lang-name\\" style=\\"font:400 11px 'Segoe UI', Ubuntu, Sans-Serif\\" class=\\"text-fill\\" x=\\"15\\" y=\\"10\\">Other - 19 mins</text></g><g data-testid=\\"undefined\\" transform=\\"translate(205, 25)\\"><circle data-testid=\\"lang-color\\" cx=\\"5\\" cy=\\"6\\" r=\\"5\\" fill=\\"#2b7489\\"></circle><text data-testid=\\"lang-name\\" style=\\"font:400 11px 'Segoe UI', Ubuntu, Sans-Serif\\" class=\\"text-fill\\" x=\\"15\\" y=\\"10\\">TypeScript - 1 min</text></g></g></svg></g></svg>"
`;
